From ca020aef7341892f36888a5e319e10e601e7c0ef Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:03 -0400
Subject: [fs] nfs: Fix error handling in __nfs_pageio_add_request

Message-id: <1403798077-28799-45-git-send-email-steved@redhat.com>
Patchwork-id: 84884
O-Subject: [RHEL7.1 PATCH 44/78] NFS: Fix error handling in __nfs_pageio_add_request
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

Handle the case where nfs_create_request() returns an error.

Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upstream Id: c1109558ae0f07a3434357cd26203bfc7adfea75
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfs/pagelist.c b/fs/nfs/pagelist.c
index fab78d1..0b8446f 100644
--- a/fs/nfs/pagelist.c
+++ b/fs/nfs/pagelist.c
@@ -947,6 +947,8 @@ static int __nfs_pageio_add_request(struct nfs_pageio_descriptor *desc,
    subreq = nfs_create_request(req->wb_context,
      req->wb_page,
      subreq, pgbase, bytes_left);
+   if (IS_ERR(subreq))
+    goto err_ptr;
    nfs_lock_request(subreq);
    subreq->wb_offset  = offset;
    subreq->wb_index = req->wb_index;
@@ -955,6 +957,10 @@ static int __nfs_pageio_add_request(struct nfs_pageio_descriptor *desc,
 
  nfs_page_group_unlock(req);
  return 1;
+err_ptr:
+ desc->pg_error = PTR_ERR(subreq);
+ nfs_page_group_unlock(req);
+ return 0;
 }
 
 static int nfs_do_recoalesce(struct nfs_pageio_descriptor *desc)
-- 
1.7.1